src/reduced_order/pod_basis_sensitivity.cpp

```diff
@@ -0,0 +1,353 @@
+    tmp.reinit(this->fullBasis.m(), this->fullBasis.n());
+    this->fullBasis.mmult(tmp, eigenvaluesSensitivity);
+    this->eigenvaluesSqrtInverse.mmult(eigenvaluesInverse, this->eigenvaluesSqrtInverse);
+    tmp.mmult(basis_sensitivity_tmp2, eigenvaluesInverse);
+    fullBasisSensitivity.add(1, basis_sensitivity_tmp1);
+    fullBasisSensitivity.add(-0.5, basis_sensitivity_tmp2);
+
+    std::ofstream out_file("basis_sensitivity.txt");
+    unsigned int precision = 7;
+    fullBasisSensitivity.print_formatted(out_file, precision);
+
+    this->pcout << "Basis sensitivities computed." << std::endl;
+}
+
+template <int dim>
+dealii::Vector<double> SensitivityPOD<dim>::computeModeSensitivity(int k) {
+    this->pcout << "Computing mode sensitivity..." << std::endl;
+
+    //Assemble LHS: (massWeightedSolutionSnapshotsCopy - diag(eigenvalue))
+    dealii::FullMatrix<double> LHS(this->massWeightedSolutionSnapshots.m(), this->massWeightedSolutionSnapshots.n());
+    LHS = this->massWeightedSolutionSnapshotsCopy; //USE MATRIX COPY MADE BEFORE TAKING SVD!!!
+    LHS.diagadd(-1*this->massWeightedSolutionSnapshots.singular_value(k));
+
+    //Get kth eigenvector
+    dealii::Vector<double> kEigenvector(this->eigenvectors.n());
+    for(unsigned int i = 0 ; i < this->eigenvectors.n(); i++){
+        kEigenvector(i) = this->eigenvectors(i, k);
+        this->pcout << "Eigenvector " << " " << kEigenvector(i) << std::endl;
+    }
+
+    //Get eigenvalue sensitivity: kEigenvalueSensitivity = kEigenvector^T * massWeightedSensitivitySnapshots * kEigenvector
+    dealii::Vector<double> kEigenvalueSensitivity_tmp(this->eigenvectors.n());
+    massWeightedSensitivitySnapshots.Tvmult(kEigenvalueSensitivity_tmp, kEigenvector);
+    double kEigenvalueSensitivity = kEigenvalueSensitivity_tmp * kEigenvector;
+    eigenvaluesSensitivity(k, k) = kEigenvalueSensitivity;
+
+    this->pcout << "Eigenvalue" << k << " " << this->massWeightedSolutionSnapshots.singular_value(k) << std::endl;
+    this->pcout << "Eigenvalue sensitivity " << k << " " << kEigenvalueSensitivity << std::endl;
+
+
+    // Assemble RHS: -(massWeightedSensitivitySnapshots - diag(kEigenvalueSensitivity))*kEigenvector
+    dealii::FullMatrix<double> RHS_tmp(massWeightedSensitivitySnapshots.m(), massWeightedSensitivitySnapshots.n());
+    RHS_tmp = massWeightedSensitivitySnapshots;
+    RHS_tmp.diagadd(-1 * kEigenvalueSensitivity);
+    dealii::Vector<double> RHS(this->eigenvectors.n());
+    RHS_tmp.vmult(RHS, kEigenvector);
+    RHS*=-1;
+
+    // Compute least squares solution
+    dealii::Householder<double> householder (LHS);
+    dealii::Vector<double> leastSquaresSolution(this->eigenvectors.n());
+    householder.least_squares(leastSquaresSolution, RHS);
+
+    // Compute eigenvector sensitivity by Gram Schmidt orthogonalization: kEigenvectorSensitivity = leastSquaresSolution - gamma*kEigenvector
+    double gamma = leastSquaresSolution * kEigenvector;
```

Comment:
My guess is that gamma is clear from:
/* Reference for POD basis sensitivity computation:
    "Local improvements to reduced-order models using sensitivity analysis of the proper orthogonal decomposition"
    Alexander Hay, Jeffrey T. Borgaard, Dominique Pelletier
    J. Fluid Mech. (2009)
    */

If it's well-defined in there, it's fine. Otherwise, consider renaming.

Reply:
Yeah, its well defined in the paper. It's really just a coefficient.